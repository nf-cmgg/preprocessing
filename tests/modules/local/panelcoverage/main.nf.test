nextflow_process {

    name "Test Process PANELCOVERAGE"
    script "modules/local/panelcoverage/main.nf"
    process "PANELCOVERAGE"

    tag "modules"
    tag "modules/local"
    tag "modules/local/panelcoverage"

    test("test") {

        when {
            process {
                """
                input[0] = [
                    [id: "test", single_end: false ],
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/bed/sample1.per-base.bed.gz", checkIfExists:true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/bed/sample1.per-base.bed.gz.csi", checkIfExists:true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/genelists/genelist_chr21_per_exon.bed",checkIfExists:true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
