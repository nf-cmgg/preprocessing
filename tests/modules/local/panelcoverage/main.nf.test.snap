{
    "test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_genelist_chr21_per_exon.mosdepth.region.dist.txt:md5,b29a7f12cef3be13215923edf6dde674"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2452395e283f5046f40bee1d26a09121"
                ],
                "regiondist": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_genelist_chr21_per_exon.mosdepth.region.dist.txt:md5,b29a7f12cef3be13215923edf6dde674"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2452395e283f5046f40bee1d26a09121"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-12T14:47:04.895941"
    }
}